### PR TITLE
Add dedicated endpoint for conversation participants

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR introduces a specialized endpoint to retrieve the conversation participants, which includes both users and agents. By doing so, we can finally eliminate the need to fetch the conversation's content to render the conversation.

### CODE REVIEW

File: front/components/assistant/conversation/ConversationParticipants.tsx [hunk 1]
```diff
@@ -1,59 +1,36 @@
...
+}: ConversationParticipantsProps) {
+  const { conversationParticipants } = useConversationParticipants({
+    conversationId,
+    workspaceId: owner.sId,
   });
-  const userParticipants = Array.from(userParticipantsMap.values());
-  const agentParticipants = Array.from(agentParticipantsMap.values());
+
+  if (!conversationParticipants) {
+    return <></>;
```
Comment (fontanierh) [inline front/components/assistant/conversation/ConversationParticipants.tsx:22]: is this different / better / worse VS returning `null` ?

File: front/lib/api/assistant/participants.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  return agents.map((a) => ({
+    configurationId: a.sId,
+    name: a.name,
+    pictureUrl: a.pictureUrl,
+  }));
+}
+
+export async function fetchConversationParticipants(
+  auth: Authenticator,
+  conversationId: string
```
Comment (spolu) [inline front/lib/api/assistant/participants.ts:81]: Let's pass `ConverstationWithoutContentType` since we have it ?

**Review by spolu — COMMENTED** [review]
Left a few comments but generally looks good :+1:

File: front/components/assistant/conversation/ConversationParticipants.tsx [hunk 3]
```diff
@@ -1,59 +1,36 @@
...
+  const { agents, users } = conversationParticipants;
 
   return (
     <div className="flex gap-6">
       <Avatar.Stack
         size="sm"
-        nbMoreItems={
-          agentParticipants.length > 4 ? agentParticipants.length - 4 : 0
-        }
+        nbMoreItems={agents.length > 4 ? agents.length - 4 : 0}
```
Comment (fontanierh) [inline front/components/assistant/conversation/ConversationParticipants.tsx:31]: nit: is this the same as `Math.max(agents.length - 4, 0)` ?

File: front/lib/api/assistant/participants.ts (hunk 2 above)
Comment (flvndvd) [reply front/lib/api/assistant/participants.ts:81]: Good catch!

File: front/components/assistant/conversation/ConversationParticipants.tsx (hunk 3 above)
Comment (flvndvd) [reply front/components/assistant/conversation/ConversationParticipants.tsx:31]: Yep, will refactor.

File: front/lib/api/assistant/participants.ts [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+      }
+
+      return acc;
+    },
+    { agentConfigurationIds: new Set(), userIds: new Set() }
+  );
+
+  const [users, agents] = await Promise.all([
+    fetchAllUsersById([...userIds]),
+    fetchAllAgentsById(auth, [...agentConfigurationIds]),
```
Comment (fontanierh) [inline front/lib/api/assistant/participants.ts:108]: would this be kind of the same as using the `conversationId` [view](https://github.com/dust-tt/dust/blob/ae2c11ce26125abb47ea99ae467ebfd4a3903a35/types/src/front/assistant/agent.ts#L126) ?

**Review by fontanierh — APPROVED** [review]
small question about agents fetching but LGTM

File: front/lib/api/assistant/participants.ts (hunk 4 above)
Comment (flvndvd) [reply front/lib/api/assistant/participants.ts:108]: Yep, will take a look and fix in another PR.

**Pull request merged**